src/test/test_event.py

```diff
@@ -0,0 +1,241 @@
+        self.assertFalse(isinstance(event, GroupIdentifyEvent))
+        self.assertFalse(isinstance(event, RevenueEvent))
+        self.assertFalse("user_id" in event)
+        self.assertFalse("plan" in event)
+        self.assertFalse("device_id" in event)
+        event.load_event_options(None)
+        self.assertFalse("user_id" in event)
+        self.assertFalse("plan" in event)
+        self.assertFalse("device_id" in event)
+        event.load_event_options(event_option)
+        self.assertTrue("user_id" in event)
+        self.assertTrue("plan" in event)
+        self.assertFalse("device_id" in event)
+        identify_event = IdentifyEvent()
+        self.assertTrue(isinstance(identify_event, EventOptions))
+        self.assertTrue(isinstance(identify_event, BaseEvent))
+        self.assertTrue(isinstance(identify_event, IdentifyEvent))
+        self.assertFalse(isinstance(identify_event, GroupIdentifyEvent))
+        self.assertFalse(isinstance(identify_event, RevenueEvent))
+        self.assertEqual(constants.IDENTIFY_EVENT, identify_event.event_type)
+        group_identify_event = GroupIdentifyEvent()
+        self.assertTrue(isinstance(group_identify_event, EventOptions))
+        self.assertTrue(isinstance(group_identify_event, BaseEvent))
+        self.assertFalse(isinstance(group_identify_event, IdentifyEvent))
+        self.assertTrue(isinstance(group_identify_event, GroupIdentifyEvent))
+        self.assertFalse(isinstance(group_identify_event, RevenueEvent))
+        self.assertEqual(constants.GROUP_IDENTIFY_EVENT, group_identify_event.event_type)
+        revenue_event = RevenueEvent()
+        self.assertTrue(isinstance(revenue_event, EventOptions))
+        self.assertTrue(isinstance(revenue_event, BaseEvent))
+        self.assertFalse(isinstance(revenue_event, IdentifyEvent))
+        self.assertFalse(isinstance(revenue_event, GroupIdentifyEvent))
+        self.assertTrue(isinstance(revenue_event, RevenueEvent))
+        self.assertEqual(constants.AMP_REVENUE_EVENT, revenue_event.event_type)
+
+    def test_callback(self):
+        def callback_func(event, code, message=None):
+            self.assertEqual(event.event_type, "test_event")
+            self.assertEqual(code, 200)
+            self.assertEqual(message, "Test Message")
+
+        test_event = BaseEvent("test_event", callback=callback_func)
+        test_event.callback(200, "Test Message")
+
+    def test_event_to_dict(self):
+        event = BaseEvent(event_type="test_event", user_id="test_user", user_properties={"email": "test@test"})
+        expect_dict = {"event_type": "test_event",
+                       "user_id": "test_user",
+                       "user_properties": {"email": "test@test"}}
+        self.assertEqual(expect_dict, event.get_event_body())
+        event["event_properties"] = {4: "4"}
+        self.assertFalse("event_properties" in event)
+        event["event_properties"] = {"test": ["4", [5, 6]]}
```

Comment:
Could this break down to multiple test cases? Seems testing various stuff in one test case

Reply:
+1 Also can we follow the test naming convention used in Stargate 
What_Condition_ExpectedResult

https://github.com/amplitude/javascript/blob/master/server/packages/stargate/test/readme.md#test-naming-convention